ENG4000-Team-A/capstone-project
# Automatically add/update machines in database

Adds automatic machine add/update background functionality.
Any new kasa smart plug on the network is added to the database as a machine.
Any existing machine with a incorrect IP is updated to the correct IP.

This runs every 10 seconds because it uses the same logic as the switch state checker.

To test adding a new machine: 
Register a new plug on the network with the Kasa app, or delete the existing machine corresponding to the old one.
Run the app.
Go to [http://127.0.0.1:8000/admin/ConsoleTrackerApp/machine/](http://127.0.0.1:8000/admin/ConsoleTrackerApp/machine/).
Check if a new machine was added within 10~ seconds.

To test updating a machine:
Change the machine's IP to a false one (eg. 127.0.0.1).
Check if the machine's IP was corrected within 10~seconds.

Issues: 
Windows has a bug where discovering switches might not work depending on your settings.

closes #162
